pkg/mcp/server/server.go

```diff
@@ -77,23 +77,6 @@ type watch struct {
 	// only accessed from connection goroutine
 	cancel func()
 	nonce  string // most recent nonce
-
```

Comment:
The intermediate watch::saveResponseAndSchedulePush is no longer needed with the new queue implementation. This should have probably been in the previous PR, but is fairly straightforward and removes a level of indirection.